### PR TITLE
Remove all old-style prod-related pipeline configurations, leaving just CI & OneBranch

### PR DESCRIPTION
**Pull request by hawkowl**
### Which issue this PR addresses:

Fixes https://dev.azure.com/msazure/AzureRedHatOpenShift/_workitems/edit/14282641

### What this PR does / why we need it:

Removes all Azure Pipelines code that interacts with prod, leaving just PR CI & E2E. There are some OneBranch files used by the OneBranch pipelines still, these also remain. The replacement of the rest tracks with the releases being moved to EV2/OneBranch.

Note: The pipelines that reference this will need to be deleted as well.

### Test plan for issue:

This PR should work :)

### Is there any documentation that needs to be updated for this PR?

Release documentation is already deprecated, so should be fine.

### CODE REVIEW

**Review by cadenmarchese — COMMENTED** [review]
Do we still need `.pipelines/mirror-images.yml`? Asking for these two pipelines that are still run daily:
https://msazure.visualstudio.com/AzureRedHatOpenShift/_build?definitionId=131155
https://msazure.visualstudio.com/AzureRedHatOpenShift/_build?definitionId=131089

**Comment by hawkowl** [comment]
@cadenmarchese I believe there's now ARO.Pipelines ones for that.

**Review by cadenmarchese — DISMISSED** [review]
We should create a card to migrate the two image mirroring pipelines to OneBranch as it doesn't look like we have a replacement in `ARO.Pipelines` yet. In the meantime we might break OCP release mirroring if we remove them.

File: .pipelines/templates/template-mirror-images.yml [hunk 1]
```diff
@@ -1,19 +0,0 @@
...
-- script: |
-    set -eu
-
-    export DST_AUTH=${{ parameters.dstAuth }}
-    export DST_ACR_NAME=${{ parameters.dstACRName }}
-    export SRC_AUTH_QUAY=${{ parameters.srcAuthQuay }}
-    export SRC_AUTH_REDHAT=${{ parameters.srcAuthRedhat }}
-
-    chmod +x ${{ parameters.deployerDirectory }}/aro
-    ${{ parameters.deployerDirectory }}/aro mirror
```
[the remark this answers is not in the document]
Comment (hawkowl) [reply .pipelines/templates/template-mirror-images.yml:18]: Those pipelines create EV2 manifests that then trigger a deployment -- they don't run directly in the pipeline.

[the remark this answers is not in the document]
Comment (hawkowl) [reply .pipelines/templates/template-mirror-images.yml:18]: The EV2 pipelines now mirror images successfully.

File: .pipelines/generate-release-notes-tagged.yml [hunk 2]
```diff
@@ -1,42 +0,0 @@
-# Azure DevOps Pipeline for generating release notes
```
Comment (SudoBrendan) [inline .pipelines/generate-release-notes-tagged.yml:1]: This is not part of our current Ev2 release process, but I think there was discussion that it should be, yes? We should have Releases in GitHub that include public release notes of what we are releasing?

**Review by jhoreman — APPROVED** [review]
This looks okay to me. I did a pass over the pipelines that are known to me and can confirm they have proper OB/EV2 counterparts now in ADO. Thanks for this cleanup Amber.

**Comment by hawkowl** [comment]
Mirror imaging is being kept in because it does not currently run in EV2 public cloud (only INT and FF), but the rest can go.

**Review by cblecker — APPROVED** [review]

**Pull request merged**